src/directional_clustering/plotters/lines.py

```diff
@@ -0,0 +1,120 @@
+from compas.geometry import normalize_vector
+from compas.geometry import add_vectors
+from compas.geometry import scale_vector
+from compas.geometry import length_vector
+
+
+__all__ = [
+    "line_sdl",
+    "vector_lines_on_faces",
+    "line_tuple_to_dict",
+    "polygon_list_to_dict"
+]
+
+
+def line_sdl(start, direction, length, both_sides=True):
+    """
+    Creates a line from a start point with a given direction and length. It will extend the line in the oposite direction as well unless both_sides is set to False.
+
+    Input
+    -----
+    start : `array`, shape(n,), n<=3
+        A point defined by XYZ coordinates.
+        If Z coordinate is not given, then the results are in 2D.
+        If Y and Z coordinates are not given, then the results are in 1D.
+    direction : `array`
+        XYZ components of the vector to create a line.
+    length : `float`
+        The length of the line.
+    both_sides : `bool`
```

Comment:
`both_sides` is an optional argument. All optionals should read argument :  `type`, optional
In this case, `both_sides` `:` ``bool`` `, optional`